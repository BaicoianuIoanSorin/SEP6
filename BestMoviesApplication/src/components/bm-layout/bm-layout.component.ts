import { Component } from '@angular/core';
import { NbSidebarService } from '@nebular/theme';
import {GENERAL_MENU_ITEMS} from "../../app/constants";

@Component({
  selector: 'bm-layout',
  template:`
      <nb-layout>
          <nb-layout-header fixed>
            <button
              nbButton
              ghost
              shape="round"
            (click)="toggle()"
            >
              <nb-icon [icon]="isToggled ? 'arrowhead-right-outline' : 'arrowhead-left-outline'"></nb-icon>
            </button>
              <a href="#" (click)="toggle()"><i class="nb-menu"></i></a>
          </nb-layout-header>
          <nb-sidebar>
            <nb-menu [items]="GENERAL_MENU_ITEMS"></nb-menu>
          </nb-sidebar>
          <nb-layout-column class="colored-column-basic">
            <router-outlet></router-outlet>
          </nb-layout-column>
      </nb-layout>
  `
})

export class BmLayoutComponent {

  GENERAL_MENU_ITEMS = GENERAL_MENU_ITEMS;
  isToggled = false;
  constructor(private sideBarService: NbSidebarService) {
  }

  toggle() {
    this.isToggled = !this.isToggled;
    this.sideBarService.toggle(true);
  }
}
