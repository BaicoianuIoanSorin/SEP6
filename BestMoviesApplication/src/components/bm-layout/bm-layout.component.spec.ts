import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BmLayoutComponent } from './bm-layout.component';

describe('BmLayoutComponent', () => {
  let component: BmLayoutComponent;
  let fixture: ComponentFixture<BmLayoutComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ BmLayoutComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(BmLayoutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
