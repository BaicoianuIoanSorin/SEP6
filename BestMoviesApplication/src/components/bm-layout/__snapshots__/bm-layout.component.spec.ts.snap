// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > bm-layout > bm-layout.component.spec.ts should match snapshot 1`] = `
<bm-layout
  GENERAL_MENU_ITEMS={[Function Array]}
  isToggled="false"
  sideBarService={[Function NbSidebarService]}
>
  <nb-layout>
    <div
      class="scrollable-container"
    >
      <div
        class="layout"
      >
        <nb-layout-header
          class="fixed"
          fixed=""
        >
          <nav
            class="fixed"
          >
            <button
              ghost=""
              nbbutton=""
              shape="round"
            >
              <nb-icon />
            </button>
            <a
              href="#"
            >
              <i
                class="nb-menu"
              />
            </a>
          </nav>
        </nb-layout-header>
        <div
          class="layout-container"
        >
          <nb-sidebar
            class="left expanded"
          >
            <div
              class="main-container main-container-fixed"
            >
              <div
                class="scrollable"
              >
                <nb-menu />
              </div>
            </div>
          </nb-sidebar>
          <div
            class="content"
          >
            <div
              class="columns"
            >
              <nb-layout-column
                class="colored-column-basic"
              >
                <router-outlet />
              </nb-layout-column>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nb-layout>
</bm-layout>
`;
